aio/content/examples/toh-pt5/src/app/hero.service.ts

```diff
@@ -12,16 +12,17 @@ export class HeroService {
   constructor(private messageService: MessageService) { }
 
   getHeroes(): Observable<Hero[]> {
-    // TODO: send the message _after_ fetching the heroes
+    const heroes = of(HEROES);
     this.messageService.add('HeroService: fetched heroes');
-    return of(HEROES);
+    return heroes;
   }
 
   // #docregion getHero
   getHero(id: number): Observable<Hero> {
-    // TODO: send the message _after_ fetching the hero
+    // TODO: Handle failing to find a hero with the specified `id`.
+    const hero = HEROES.find(h => h.id === id) as Hero;
```

Comment:
May be we can update the comment to something like this?
```suggestion
    // In order to keep the example app simple, we assume that
    // a Hero with the provided id is always present in the list.
    const hero = HEROES.find(hero => hero.id === id) as Hero;
```